catanatron_rust/src/state.rs

```diff
@@ -333,13 +352,84 @@ impl State {
             let mut current_path = Vec::new();
             let mut best_path = Vec::new();
 
-            self.dfs_longest_path(start_node, None, connected_node_set, color, &mut current_path, &mut best_path);
+            self.dfs_longest_path(
+                start_node,
+                None,
+                connected_node_set,
+                color,
+                &mut current_path,
+                &mut best_path,
+            );
             if best_path.len() > overall_best_path.len() {
                 overall_best_path = best_path;
             }
         }
         overall_best_path
     }
+
+    pub fn add_dev_card(&mut self, color: u8, card_idx: usize) {
+        self.vector[player_devhand_slice(self.config.num_players, color)][card_idx] += 1;
+    }
+
+    pub fn get_dev_card_count(&self, color: u8, card_idx: usize) -> u8 {
+        self.vector[player_devhand_slice(self.config.num_players, color)][card_idx]
+    }
+
+    pub fn get_played_dev_card_count(&self, color: u8, card_idx: usize) -> u8 {
+        self.vector[player_played_devhand_slice(self.config.num_players, color)][card_idx]
+    }
+
+    pub fn add_played_dev_card(&mut self, color: u8, card_idx: usize) {
+        self.vector[player_played_devhand_slice(self.config.num_players, color)][card_idx] += 1;
+    }
+
+    pub fn remove_dev_card(&mut self, color: u8, card_idx: usize) {
+        self.vector[player_devhand_slice(self.config.num_players, color)][card_idx] -= 1;
+    }
+
+    pub fn set_has_played_dev_card(&mut self) {
+        self.vector[HAS_PLAYED_DEV_CARD] = 1;
+    }
+
+    pub fn set_is_moving_robber(&mut self) {
+        self.vector[IS_MOVING_ROBBER_INDEX] = 1;
+    }
+
+    pub fn clear_is_moving_robber(&mut self) {
+        self.vector[IS_MOVING_ROBBER_INDEX] = 0;
+    }
+
+    pub fn bank_has_resource(&self, resource: u8) -> bool {
+        self.vector[BANK_RESOURCE_SLICE][resource as usize] > 0
+    }
+
+    pub fn take_from_bank_give_to_player(&mut self, color: u8, resource: u8) {
+        let resource_idx = resource as usize;
+        self.vector[BANK_RESOURCE_SLICE][resource_idx] -= 1;
+        self.get_mut_player_hand(color)[resource_idx] += 1;
+    }
+
+    pub fn take_from_player_give_to_bank(&mut self, color: u8, resource: u8, amount: u8) {
+        let resource_idx = resource as usize;
+        self.get_mut_player_hand(color)[resource_idx] -= amount;
+        self.vector[BANK_RESOURCE_SLICE][resource_idx] += amount;
+    }
+
+    pub fn get_player_resource_count(&self, color: u8, resource: u8) -> u8 {
+        self.get_player_hand(color)[resource as usize]
+    }
+
+    pub fn take_from_player_give_to_player(
```

Comment:
NIT: Feel free to: `from_bank_to_player`, `from_player_to_bank`, and `from_player_to_player`. I think the rest of the signature completes getting the message across. 👍